src/seeds.bracelet.cpp

```diff
@@ -0,0 +1,141 @@
+
+    utils::check_asset(quantity);
+
+    name target = from;
+
+    check_user(target);
+
+    init_balance(target);
+    init_balance(_self);
+
+    add_balance(target, quantity);
+
+    _deposit(quantity);
+  }
+
+}
+
+ACTION bracelet::freeze (name account) {
+  require_auth(account);
+
+  auto bitr = balances.find(account.value);
+  check(bitr != balances.end(), "bracelet: user " + account.to_string() + " has no balance entry");
+
+  balances.modify(bitr, _self, [&](auto & item){
+    item.is_freezed = true;
+  });
+}
+
+ACTION bracelet::unfreeze (name account) {
+  require_auth(account);
+
+  auto bitr = balances.find(account.value);
+  check(bitr != balances.end(), "bracelet: user " + account.to_string() + " has no balance entry");
+
+  balances.modify(bitr, _self, [&](auto & item){
+    item.is_freezed = false;
+  });
+}
+
+ACTION bracelet::withdraw (name account, asset quantity) {
+  require_auth(account);
+  string memo = "";
+  sub_balance(account, quantity);
+  _transfer(account, quantity, memo);
+}
+
+ACTION bracelet::transfer (name from, name to, asset quantity, string memo) {
+  require_auth(permission_level(from, "bracelet"_n));
+  check_freeze(from);
+  sub_balance(from, quantity);
+  _transfer(to, quantity, memo);
+}
+
+void bracelet::init_balance (name account) {
+  auto bitr = balances.find(account.value);
+  if (bitr == balances.end()) {
+    balances.emplace(_self, [&](auto & item){
+      item.account = account;
+      item.balance = asset(0, utils::seeds_symbol);
+      item.is_freezed = false;
+    });
+  }
+}
+
+void bracelet::add_balance (name account, asset quantity) {
+  utils::check_asset(quantity);
+
+  auto bitr = balances.find(account.value);
+  check(bitr != balances.end(), "bracelet: no balance object found for " + account.to_string());
+
+  auto bcitr = balances.find((get_self()).value);
+  check(bcitr != balances.end(), "bracelet: contract has not balance entry");
+
+  balances.modify(bitr, _self, [&](auto & item){
+    item.balance += quantity;
+  });
+
+  balances.modify(bcitr, _self, [&](auto & item){
+    item.balance += quantity;
+  });
+}
+
+void bracelet::sub_balance (name account, asset quantity) {
+  utils::check_asset(quantity);
+
+  auto bitr = balances.find(account.value);
+  check(bitr != balances.end(), "bracelet: no balance object found for " + account.to_string());
+  check(bitr -> balance.amount >= quantity.amount, "bracelet: overdrawn balance");
+
+  auto bcitr = balances.find((get_self()).value);
+  check(bcitr != balances.end(), "bracelet: contract has not balance entry");
+  
+  balances.modify(bitr, _self, [&](auto & item){
+    item.balance -= quantity;
+  });
+
+  balances.modify(bcitr, _self, [&](auto & item){
+    item.balance -= quantity;
+  });
+}
+
+void bracelet::_deposit (asset quantity) {
+  check_asset(quantity);
+
+  token::transfer_action action{contracts::token, {_self, "active"_n}};
+  action.send(_self, contracts::bank, quantity, "");
```

Comment:
Looks like the memo is empty here on this transfer, should it mention it's a "bracelet deposit"?

Reply:
ok, I will change it. What about the memo in the `withdraw` action?, I left it empty as well for the moment

Reply:
Perfect, that's good for now. :)